Panic: nil pointer dereference in dataSourceNetboxDevicesRead when DeviceType is nil

## Terraform Version

```
Terraform v1.12.2
+ provider registry.terraform.io/e-breuninger/netbox v5.3.0
```

NetBox version: 4.4.5

## Affected Resource(s)

- `data.netbox_devices`

## Panic Output

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x125d8d1]

goroutine 128 [running]:
github.com/e-breuninger/terraform-provider-netbox/netbox.dataSourceNetboxDevicesRead(0xc000358c60, {0x17a8640?, 0xc000602570})
	github.com/e-breuninger/terraform-provider-netbox/netbox/data_source_netbox_devices.go:258 +0x1651
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.40.0/helper/schema/resource.go:854 +0x15f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.40.0/helper/schema/resource.go:1088 +0x13e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.40.0/helper/schema/grpc_provider.go:1994 +0x6af
```

The full Terraform error:

```
Error: Plugin did not respond

  with data.netbox_devices.my_device[0],
  on main.tf line 10, in data "netbox_devices" "my_device":
  10: data "netbox_devices" "my_device" {

The plugin encountered an error, and failed to respond to the
plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain
more details.
```

## Terraform Configuration

```hcl
data "netbox_devices" "my_device" {
  filter {
    name  = "name"
    value = "my-device-01"
  }
}
```

## Root Cause

Looking at `data_source_netbox_devices.go` around line 258, the code has:

```go
if device.DeviceType != nil {
    mapping["device_type_id"] = device.DeviceType.ID
}
if device.DeviceType.Manufacturer != nil {
    mapping["manufacturer_id"] = device.DeviceType.Manufacturer.ID
}
if device.DeviceType.Model != nil {
    mapping["model"] = *device.DeviceType.Model
}
```

The second and third `if` statements access `device.DeviceType.Manufacturer` and `device.DeviceType.Model` **without** first checking that `device.DeviceType` is not nil. If a device has no `DeviceType` set (or the API response does not include it), the provider panics with a nil pointer dereference.

## Suggested Fix

Nest the `Manufacturer` and `Model` checks inside the `DeviceType != nil` block:

```go
if device.DeviceType != nil {
    mapping["device_type_id"] = device.DeviceType.ID
    if device.DeviceType.Manufacturer != nil {
        mapping["manufacturer_id"] = device.DeviceType.Manufacturer.ID
    }
    if device.DeviceType.Model != nil {
        mapping["model"] = *device.DeviceType.Model
    }
}
```

## Additional Context

- Provider v5.1.0 works without this issue (upgraded from 5.1.0 to 5.3.0).
- This is likely related to #864 (better error checking in read functions) which was included in v5.3.0 but did not cover `data_source_netbox_devices.go`.
- The crash kills the provider process entirely, causing cascading "Plugin did not respond" and "Request cancelled" errors for other unrelated resources in the same plan run.

--- a/netbox/data_source_netbox_devices.go
+++ b/netbox/data_source_netbox_devices.go
@@ -296,12 +296,12 @@ func dataSourceNetboxDevicesRead(d *schema.ResourceData, m interface{}) error {
 		mapping["device_id"] = device.ID
 		if device.DeviceType != nil {
 			mapping["device_type_id"] = device.DeviceType.ID
-		}
-		if device.DeviceType.Manufacturer != nil {
-			mapping["manufacturer_id"] = device.DeviceType.Manufacturer.ID
-		}
-		if device.DeviceType.Model != nil {
-			mapping["model"] = *device.DeviceType.Model
+			if device.DeviceType.Manufacturer != nil {
+				mapping["manufacturer_id"] = device.DeviceType.Manufacturer.ID
+			}
+			if device.DeviceType.Model != nil {
+				mapping["model"] = *device.DeviceType.Model
+			}
 		}
 		if device.Name != nil {
 			mapping["name"] = *device.Name

--- a/netbox/data_source_netbox_devices_nil_device_type_test.go
+++ b/netbox/data_source_netbox_devices_nil_device_type_test.go
@@ -1,0 +1,55 @@
+package netbox
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// TestDataSourceNetboxDevicesReadNilDeviceType verifies that reading a device
+// whose device_type is null does not panic. NetBox can return devices without a
+// device type, and the data source must skip the device_type-derived attributes
+// instead of dereferencing a nil pointer.
+func TestDataSourceNetboxDevicesReadNilDeviceType(t *testing.T) {
+	device := minimalDeviceJSON(1, "device-without-type")
+	device["device_type"] = nil
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path != "/api/dcim/devices/" {
+			http.NotFound(w, r)
+			return
+		}
+		w.Header().Set("Content-Type", "application/json")
+		json.NewEncoder(w).Encode(map[string]interface{}{
+			"count":    1,
+			"next":     nil,
+			"previous": nil,
+			"results":  []map[string]interface{}{device},
+		})
+	}))
+	defer ts.Close()
+
+	cfg := Config{APIToken: "test-token", ServerURL: ts.URL}
+	client, err := cfg.Client()
+	if err != nil {
+		t.Fatalf("failed to create client: %v", err)
+	}
+
+	state := &providerState{NetBoxAPI: client}
+	d := schema.TestResourceDataRaw(t, dataSourceNetboxDevices().Schema, map[string]interface{}{})
+
+	if err := dataSourceNetboxDevicesRead(d, state); err != nil {
+		t.Fatalf("dataSourceNetboxDevicesRead returned error: %v", err)
+	}
+
+	devices := d.Get("devices").([]interface{})
+	if len(devices) != 1 {
+		t.Fatalf("got %d devices, want 1", len(devices))
+	}
+	if mapping := devices[0].(map[string]interface{}); mapping["device_type_id"] != 0 {
+		t.Errorf("expected no device_type_id for device without device type, got %v", mapping["device_type_id"])
+	}
+}

